orttraining/orttraining/python/training/debug.py

```diff
@@ -0,0 +1,33 @@
+
+import numpy as np
+import os
+import sys
+import torch
+
+from numpy.testing import assert_allclose
+
+# Inputs are the state_dicts for each of the model (dictionary for each layer and weights)
+def compare_onnx_weights(model_a, model_b, verbose=False, rtol=1e-4):
+    r"""Compare onnx weights method
+    Compares the weights of two different ONNX models and throws an error when they diverge
+    Args:
+        model_a, model_b: The state_dicts of two ONNX models containing a dictionary 
+    of the weights for each layer of the respective model.
+        verbose: Indicates if the max absolute difference for each layer should be
+    calculated and printed for debug information. Default: False.
+        rtol: Tolerance for divergence. Default: 1e-4.
+
+    """
+    assert isinstance(model_a, dict) and isinstance(model_b, dict)
+
```

Comment:
Should the dict have the same size? Any other property that we should *assert* before testing?